xml-format.md

```diff
@@ -0,0 +1,207 @@
+data type mappings for CloudEvents context attributes.
+
+* The [Data](#3-data) section defines the container for the data portion of a
+CloudEvent.
+
+* The [Envelope](#4-envelope) section defines an XML container for CloudEvents
+attributes and an associated media type.
+
+* The [Batch](#5-xml-batch-format) section describes how multiple CloudEvents
+may be packaged into a single XML document.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Approach
+
+This specification promotes an attribute heavy XML document style, such a model
+presents advantages in terms of parsing/processing along with a more compact
+XML document representation.
+
+Attribute and Element names are deliberately terse for performance reasons.
+
+Preservation of attribute value type information is supported allowing
+custom extensions to be communicated without value type loss.
+
+The model is formalized by this [XML Schema][xml-spec].
+
+## 2. Attributes
+
+The CloudEvents type system is mapped to XML schema types as follows :
+
+| CloudEvents   | XML | Type Tag |
+| ------------- | ---------------------------------------------------------------------- | -- |
+| Boolean       | [xs:boolean][xml-primitives] | B |
+| Integer       | [xs:int][xml-primitives] | I |
+| String        | [xs:string][xml-primitives] | S |
+| Binary        | [xs:base64Binary][xml-primitives] | BI |
+| URI           | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986]| U |
+| URI-reference | [xs:anyURI][xml-primitives] following [RFC 3986][rfc3986] | UR |
+| Timestamp     | [xs:dateTime][xml-primitives] following [RFC 3339][rfc3339] | T |
+
+Each concrete CloudEvent context attribute is represented using an XML element with the following
+shape:
+
+``` xml
+<Attr n="id" v="XXX-YYY-ZZZ-0000" t="S">
+```
+
+where :
+
+* _n_ is the _name_ of the context attribute
+* _v_ is the _value_ of the context attribute
+* _t_ is the _type_ of the attribute value acording to the _Tag_ defined in the table above.
+
+See section below on special handling for `specversion`
+
+## 2.1 Specification Version
+
+To better support streaming style processing the specification version
+(`specversion` context attribute) statement is communicated in an explicit element.
+
+``` xml
+<Ver>1.0</Ver>
+```
+
+## 3. Data
+
+The `Data` portion of a CloudEvent is contained is a `<Data></Data>` element and supports
+Binary, Text, and XML Data. Each represention is wrapped in a discriminating element:
+
+### Binary Data
+
+Carried in an element with an expected type of `xs:base64Binary`
+
+``` xml
+<Data>
+    <Bin>... base64 encoded string ...</Bin>
+</Data>
+```
+
+### Text Data
+
+Carried in an element with an expected type of `xs:string`
+
+``` xml
+<Data>
+    <Txt>Hello World</Txt>
```

Comment:
I'm a bit concerned that XML whitespace rules could lose data unless SDK authors are really careful. I suspect we should give a warning here with an example.

Reply:
Could you clarify - sorry, it really has been that long ;-)

Reply:
I suspect that string data with a value of "First line\n\n\nFourth line" might end up as "First line Fourth line" after XML whitespace handling. I may be wrong, and it would certainly depend on the XML API being used. Happy to look into it further.

Reply:
I believe this should be resolved ... the comment is over a year old so if still a concern we can re-create it.